notion/block.py

```diff
@@ -735,10 +735,23 @@ class CollectionViewBlockViews(Children):
     child_list_key = "view_ids"
 
     def _get_block(self, view_id):
-        return self._client.get_collection_view(
+
+        view = self._client.get_collection_view(
             view_id, collection=self._parent.collection
         )
 
+        i = 0
+        while view is None:
+            i += 1
+            if i > 20:
+                return None
+            time.sleep(0.1)
+            view = self._client.get_collection_view(
+                view_id, collection=self._parent.collection
+            )
```

Comment:
When testing, this doesn't seem to resolve the issue with the `None` views. It doesn't seem like a race-condition (I even bumped the iterations to 70). With this change, my _add_task_ takes 16~ seconds. If I remove this block of code then it takes about 2.5~ seconds (and still works).

As mentioned in my issue, when I print the `print(collection.parent.views)` that last view resolved and is a valid view. I feel like something else is up here 🤔.

For the time being, the following did resolve this for me due to the last view being valid:

```
# In collection.py L154
                if view is None or isinstance(view, CalendarView):
```

-----

I'm all on board for working with you to resolve this though.

> One guess is that perhaps there were previously some other views, and they got deleted, but their IDs were left on the CollectionViewBlock's view_ids field. So when it builds the list of views, it tries to fetch those views and doesn't find them, and hence returns None

This is something I can dig into, just need a bit more time to explore this (although the amount of returned views matches the number of views I have in my collection on Notion).

Reply:
I've merged this (and some other fixes) to master now (not yet released, still working through the PRs and doing testing). If you're able to replicate on the latest master, let's dig in further.